cloud/pkg/edgecontroller/controller/upstream.go

```diff
@@ -667,6 +676,92 @@ func (uc *UpstreamController) processServiceAccountToken() {
 	}
 }
 
+func (uc *UpstreamController) processLease() {
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("stop process lease")
+			return
+		case msg := <-uc.leaseChan:
+			content, err := msg.GetContentData()
+			if err != nil {
+				klog.Errorf("failed to get content data from lease message: %s, %v", msg.String(), err)
+				continue
+			}
+			lease := &coordinationv1.Lease{}
+			if err = json.Unmarshal(content, lease); err != nil {
+				klog.Errorf("unmarshal lease query request failed err: %v", err)
+				continue
+			}
+			nodeID, err := messagelayer.GetNodeID(msg)
+			if err != nil {
+				klog.Errorf("failed to get nodeID of lease msg %s, %v", msg.String(), err)
+				continue
+			}
+			resource, err := messagelayer.BuildResource(nodeID, lease.Namespace, model.ResourceTypeLease, lease.Name)
```

Comment:
I think we can use the parent resource for the response message

Reply:
Do you mean we can use `msg.Resource` as the `reponse.Resource`?

Reply:
Right